scripts/autolabel.lean

```diff
@@ -0,0 +1,160 @@
+/-
+Copyright (c) 2024 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jon Eugster, Damiano Testa
+-/
+import Lean.Elab.Command
+
+/-!
+# Automatic labelling of PRs
+
+This file contains the script to automatically assign a GitHub label to a PR.
+
+`AutoLabel.mathlibLabels` contains an assignment of github labels to folders inside
+the mathlib repository. The sctript uses `git diff` to determine which files
+a PR modifies and then finds all labels that should be added based on these changes.
+
+For the time being, the script only adds a label if it finds a single unique label
+that would apply. If multiple labels are found, nothing happens.
+-/
+
+open Lean
+
+namespace AutoLabel
+
+/--
+A `Label` consists of the
+* The `label` field is the actual GitHub label.
+* The `dirs` field is the array of all "root paths" such that a modification in a file contained
+  in one of these paths should be labeled by `label`.
+* The `exclusions` field is the array of all "root paths" that are excluded, among the
+  ones that start with the ones in `dirs`.
+-/
+structure Label where
+  /-- The label name as it appears on github -/
+  label : String
+  /-- Array of paths which fall under this label. e.g. `"Mathlib" / "Algebra"` -/
+  dirs : Array System.FilePath
+  /-- Array of (sub)-paths which should be excluded -/
+  exclusions : Array System.FilePath := #[]
+  deriving BEq, Hashable
+
+/--
+Mathlib Labels and their corresponding folders. Add new labels and folders here!
+-/
+private def mathlibLabels : Std.HashSet Label := Std.HashSet.ofList [
```

Comment:
```suggestion
def mathlibLabels : Array Label := #[
```
Here is my reasoning for the changes:
* a file in `scripts` will not be imported, so `private` seems superfluous;
* `mathlibLabels` is used only once and immediately converted to an `Array`, so it might as well be an `Array`!